src/main/java/lab2/AvatarGenerator.java

```diff
@@ -1,66 +1,76 @@
+import java.net.http.HttpResponse; // handles an HTTP response
 
-public class AvatarGenerator {
+public class AvatarGenerator { // AvatarGenerator class in the 'lab2' package. This class generates and displays random avatars
 
-    public static void main(String[] args) {
+    public static void main(String[] args) { // this is the entry point for running the program
 
         try {
-            var avatarStream = AvatarGenerator.getRandomAvatarStream();
-            AvatarGenerator.showAvatar(avatarStream);
-        } catch (IOException | InterruptedException e) {
-            e.printStackTrace();
-        }
 
+            var avatarStream = AvatarGenerator.getRandomAvatarStream(); // Call to getRandomAvatarStream to fetch a random avatar
+            // avatarString returns an InputStream, object reference type
+            AvatarGenerator.showAvatar(avatarStream); // Call to showAvatar with the fetched InputStream to display the avatar
+        } catch (IOException | InterruptedException e) { // Catch IOException or InterruptedException
+            e.printStackTrace(); // Print the stack trace for debugging
+        }
     }
 
     public static InputStream getRandomAvatarStream() throws IOException, InterruptedException {
-        // Pick a random style
-        String[] styles = { "adventurer", "adventurer-neutral", "avataaars", "big-ears", "big-ears-neutral", "big-smile", "bottts", "croodles", "croodles-neutral", "fun-emoji", "icons", "identicon", "initials", "lorelei", "micah", "miniavs", "open-peeps", "personas", "pixel-art", "pixel-art-neutral" };
-        var style = styles[(int)(Math.random() * styles.length)];
+        // generates a random avatar stream. Throws IOException and InterruptedException
+
+        String[] styles = { "adventurer", "adventurer-neutral", "avataaars", "big-ears", "big-ears-neutral", "big-smile"
+                , "bottts", "croodles", "croodles-neutral", "fun-emoji", "icons", "identicon", "initials", "lorelei",
+                "micah", "miniavs", "open-peeps", "personas", "pixel-art", "pixel-art-neutral" }; // styles is an array of strings
+        // Randomly selects one one avatar style from the array "styles"
+        var style = styles[(int)(Math.random() * styles.length)]; // "style" is a String
 
-        // Generate a random seed
-        var seed = (int)(Math.random() * 10000);
+        // Generates the random seed for the avatar HTTP request
+        var seed = (int)(Math.random() * 10000); // 'seed' is a primitive type, integer value
 
-        // Create an HTTP request for a random avatar
+        // Create an HTTP request URI to fetch a random avatar with a seed
         var uri = URI.create("https://api.dicebear.com/9.x/%s/png?seed=%d".formatted(style, seed));
+        // uri returns a java.net.URI type, reference type
+
+        // Create an HttpRequest object with the generated URI
         var request = HttpRequest.newBuilder(uri).build();
```

Comment:
You need to discuss what newBuilder and build do, and whether they are class/instance methods